### PR TITLE
Add getting CRD from XRD as well as claim CRD.

### LINKED ISSUE #89: Get CustomResourceDefinition from CompositeResourceDefinition
### What problem are you facing?

The only way to get a CustomResourceDefinition is either to get it from the list of all resource definitions (`customResourceDefinitions.nodes`) or directly by `id` from `kubernetesResource(id: $id)`. The list of all CRDs can be quite long and if the user is just wanted the CRDs the correspond to the `CompositeResourceDefinition`s fetching the whole is and expensive way to get them. The `CompositeResourceDefinition` does offer the `apiVersion` and `kind` of composite and claim CRDs via `status.controllers.compositeResourceType` and `status.controllers.compositeResourceClaimType` respectively but there is no way to fetch a `CRD` from that info.

### How could Upbound help solve your problem?

If a `CompositeResourceDefinition` offered fields that returned the CRD for the claim and the composite resource this would be a much smaller list and a more performant way of getting these details. Also, it would be nice to be able to fetch a CRD from `kind` and `apiVersion` but this is not strictly needed currently.

### PR DESCRIPTION
**Pull request by thephred**
### Description of your changes

Add getting CRD from XRD as well as claim CRD.

Resolves #89

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

Ran `make` and `make reviewable`

### CODE REVIEW

File: internal/graph/resolvers/apiextensions_test.go [hunk 1]
```diff
@@ -44,6 +44,254 @@ var (
...
+				obj: &model.CompositeResourceDefinition{Spec: &model.CompositeResourceDefinitionSpec{Names: &model.CompositeResourceDefinitionNames{}}},
+			},
+			want: want{
+				errs: gqlerror.List{
+					gqlerror.Errorf(errors.Wrap(errBoom, errGetClient).Error()),
+				},
+			},
+		},
+		"GetError": {
+			reason: "If we can't get a client we should add the error to the GraphQL context and return early.",
```
Comment (hasheddan) [inline internal/graph/resolvers/apiextensions_test.go:104]:
```suggestion
			reason: "If we can't get a CompositeResourceDefinition we should add the error to the GraphQL context and return early.",
```

File: internal/graph/resolvers/apiextensions_test.go [hunk 2]
```diff
@@ -44,6 +44,254 @@ var (
...
+				obj: &model.CompositeResourceDefinition{Spec: &model.CompositeResourceDefinitionSpec{ClaimNames: &model.CompositeResourceDefinitionNames{}}},
+			},
+			want: want{
+				errs: gqlerror.List{
+					gqlerror.Errorf(errors.Wrap(errBoom, errGetClient).Error()),
+				},
+			},
+		},
+		"GetError": {
+			reason: "If we can't get a client we should add the error to the GraphQL context and return early.",
```
Comment (hasheddan) [inline internal/graph/resolvers/apiextensions_test.go:228]:
```suggestion
			reason: "If we can't get a CompositeResourceDefinition we should add the error to the GraphQL context and return early.",
```

**Review by hasheddan — APPROVED** [review]
@thephred two minor nitpicks but LGTM!

**Review by jmaduroUpbound — APPROVED** [review]
lgtm 🐬

**Pull request merged**